packages/kit/src/runtime/server/page/index.js

```diff
@@ -79,13 +79,38 @@ export async function render_page(event, route, page, options, state, resolve_op
 		// it's crucial that we do this before returning the non-SSR response, otherwise
 		// SvelteKit will erroneously believe that the path has been prerendered,
 		// causing functions to be omitted from the manifesst generated later
-		const should_prerender = get_option(nodes, 'prerender') ?? false;
+		const should_prerender = get_option(nodes, 'prerender');
+
+		console.log('SHOULD', should_prerender, event.url.href);
 		if (should_prerender) {
 			const mod = leaf_node.server;
 			if (mod && mod.actions) {
 				throw new Error('Cannot prerender pages with actions');
 			}
 		} else if (state.prerendering) {
+			// Try to render the shell when ssr is false.
+			// People can opt out of this behavior by explicitly setting prerender to false
+			if (
+				should_prerender !== false &&
+				get_option(nodes, 'ssr') === false &&
+				!leaf_node.server?.actions
+			) {
+				return await render_response({
```

Comment:
duplicated to code a little further below, but this way it's more readable (you don't need to hold all these conditions in your head until you arrive at the other condition)